cloudevents/pydantic/event.py

```diff
@@ -0,0 +1,240 @@
+
+    subject: typing.Optional[pydantic.constr(min_length=1)] = pydantic.Field(
+        title="Event Subject",
+        description=(
+            "This describes the subject of the event in the context of the event producer (identified by source). "
+            "In publish-subscribe scenarios, a subscriber will typically subscribe to events emitted by a source, "
+            "but the source identifier alone might not be sufficient as a qualifier for any specific event if "
+            "the source context has internal sub-structure.\n"
+            "\n"
+            "Identifying the subject of the event in context metadata (opposed to only in the data payload) is "
+            "particularly helpful in generic subscription filtering scenarios where middleware is unable "
+            "to interpret the data content. In the above example, the subscriber might only be interested in blobs "
+            "with names ending with '.jpg' or '.jpeg' and the subject attribute allows for constructing "
+            "a simple and efficient string-suffix filter for that subset of events."
+        ),
+    )
+    datacontenttype: typing.Optional[
+        event_attribute_types.ContentType
+    ] = pydantic.Field(
+        title="Event Data Content Type",
+        description=(
+            "Content type of data value. This attribute enables data to carry any type of content, "
+            "whereby format and encoding might differ from that of the chosen event format."
+        ),
+    )
+    dataschema: typing.Optional[
+        event_attribute_types.NonEmptyAbsoluteURI
+    ] = pydantic.Field(
+        title="Event Data Schema",
+        description=(
+            "Identifies the schema that data adheres to. "
+            "Incompatible changes to the schema SHOULD be reflected by a different URI"
+        ),
+    )
+
+    def __init__(self, attributes=None, data=None, *args, **kwargs):
+        if attributes:
+            # to prevent API complexity and confusion
+            assert (
+                len(kwargs) == 0
+            ), "legacy attributes with direct initialization is incompatible"
+            attributes = {k.lower(): v for k, v in attributes.items()}
+            kwargs.update(attributes)
+        super(CloudEvent, self).__init__(data=data, *args, **kwargs)
+
+    class Config:
+        extra: str = "allow"  # this is the way we implement extensions
+        schema_extra = {
+            "examples": [
+                {
+                    "specversion": "1.0",
+                    "type": "com.github.pull_request.opened",
+                    "source": "https://github.com/cloudevents/spec/pull",
+                    "subject": "123",
+                    "id": "A234-1234-1234",
+                    "time": "2018-04-05T17:31:00Z",
+                    "comexampleextension1": "value",
+                    "comexampleothervalue": 5,
+                    "datacontenttype": "text/xml",
+                    "data": '<much wow="xml"/>',
```

Comment:
```suggestion
                    "data": "<much wow=\"xml\"/>",
```

Reply:
black won't let me change it, it re-formats it back to this